phoenix-core/src/main/java/org/apache/phoenix/execute/BaseQueryPlan.java

```diff
@@ -353,11 +357,12 @@ public final ResultIterator iterator(final Map<ImmutableBytesPtr,ServerCache> ca
                 KeyValueSchema schema = ProjectedColumnExpression.buildSchema(dataColumns);
                 // Set key value schema of the data columns.
                 serializeSchemaIntoScan(scan, schema);
-                
-                // Set index maintainer of the local index.
-                serializeIndexMaintainerIntoScan(scan, dataTable);
-                // Set view constants if exists.
-                serializeViewConstantsIntoScan(scan, dataTable);
+                if (table.getIndexType() == IndexType.LOCAL) {
```

Comment:
Don't we need this for global uncovered ones too?

Reply:
No. We populate scan attributes for global indexes at the constructor of the TableResultIterator using ScanUtil#setScanAttributesForClient in general when the table to be scanned is a global index table.